pkg/docker/config/config_linux.go

```diff
@@ -0,0 +1,81 @@
+package config
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/containers/image/pkg/keyctl"
+)
+
+func getAuthFromKernelKeyring(registry string) (string, string, error) {
+	userkeyring, err := keyctl.UserKeyring()
+
+	if err != nil {
+		return "", "", err
+	}
+	key, err := userkeyring.Search(genDescription(registry))
+	if err != nil {
+		return "", "", err
+	}
+	authData, err := key.Get()
+	if err != nil {
+		return "", "", err
+	}
+	parts := strings.SplitN(string(authData), ":", 2)
+	if len(parts) != 2 {
+		return "", "", nil
+	}
+	return parts[0], parts[1], nil
+}
+
+func deleteAuthFromKernelKeyring(registry string) error {
+	userkeyring, err := keyctl.UserKeyring()
+
+	if err != nil {
+		return err
+	}
+	key, err := userkeyring.Search(genDescription(registry))
+	if err != nil {
+		return err
+	}
+	return key.Unlink()
+}
+
+func setAuthToKernelKeyring(registry, username, password string) error {
+	keyring, err := keyctl.SessionKeyring()
+
+	if err == nil {
+		id, err := keyring.Add(genDescription(registry), []byte(fmt.Sprintf("%s:%s", username, password)))
+		if err != nil {
+			return err
+		}
+
+		// Set the user permisson to link the key to userkeyring later
+		err = keyctl.SetPerm(id, keyctl.PermUserAll)
+		if err != nil {
+			return err
+		}
+		// link the key to userKeyring
+		userKeyring, err := keyctl.UserKeyring()
+		if err != nil {
+			fmt.Println("user kring", err)
```

Comment:
This looks like an in-development artefact? If we want to keep them, we need to use `logrus` and be a bit more descriptive to make it easier to spot when reading the wall of debugging messages in our stack.